nrf-hal-common/src/delay.rs

```diff
@@ -27,28 +27,46 @@ impl Delay {
 
 impl DelayMs<u32> for Delay {
     fn delay_ms(&mut self, ms: u32) {
-        self.delay_us(ms * 1_000);
+        DelayNs::delay_ms(self, ms);
     }
 }
 
 impl DelayMs<u16> for Delay {
     fn delay_ms(&mut self, ms: u16) {
-        self.delay_ms(u32(ms));
+        DelayNs::delay_ms(self, ms.into());
     }
 }
 
 impl DelayMs<u8> for Delay {
     fn delay_ms(&mut self, ms: u8) {
-        self.delay_ms(u32(ms));
+        DelayNs::delay_ms(self, ms.into());
     }
 }
 
 impl DelayUs<u32> for Delay {
     fn delay_us(&mut self, us: u32) {
+        DelayNs::delay_us(self, us);
+    }
+}
+
+impl DelayUs<u16> for Delay {
+    fn delay_us(&mut self, us: u16) {
+        DelayNs::delay_us(self, us.into());
+    }
+}
+
+impl DelayUs<u8> for Delay {
+    fn delay_us(&mut self, us: u8) {
+        DelayNs::delay_us(self, us.into());
+    }
+}
+
+impl DelayNs for Delay {
+    fn delay_ns(&mut self, ns: u32) {
         // The SysTick Reload Value register supports values between 1 and 0x00FFFFFF.
         const MAX_RVR: u32 = 0x00FF_FFFF;
 
-        let mut total_rvr = us * (HFCLK_FREQ / 1_000_000);
+        let mut total_rvr = ns * (HFCLK_FREQ / 1_000_000_000);
```

Comment:
`HFCLK_FREQ` is `64_000_000`, so isn't this always going to come out as zero?

Reply:
Whoops, good point. Fixed.

Reply:
That fix looks correct to me, and I can verify that the SYST-based `DelayNs` implementation works given the fix, when it didn't before.

But I'm not a maintainer of this repository, and I haven't read through this whole PR.